### ISSUE TITLE
Query: Exception with nested Select

### ISSUE THREAD

**Issue by fabiohvp** [dotnet/efcore#2817]
Hello, 
the problem is when I want to make reusable Projections for select. I'll explain.

I have the following interface:

``` c#
public interface ISelector<TModel>
        where TModel : class
    {
        Expression<Func<TModel, object>> Model { get; }
    }
```

And the folloing classes:

``` c#
public class ItemSelector : ISelector<Item>
    {
        public System.Linq.Expressions.Expression<System.Func<Item, object>> Model
        {
            get
            {
                return item => new
                {
                    Name = item.Name,
                    Resources = item.Resources.AsQueryable().Select(new ItemResourceSelector().Model)
                };
            }
        }
    }


public class ItemResourceSelector : ISelector<ItemResource>
    {
        public System.Linq.Expressions.Expression<System.Func<ItemResource, object>> Model
        {
            get
            {
                return resource => new
                {
                    Name = resource.Resource.Name,
                    Amount = resource.Amount
                };
            }
        }
    }
```

Finally, I have the following Query:

``` c#
 return DbContext.Items
                .Include(item => item.Resources.Select(resource => resource.Resource))
                .Select(new ItemSelector().Model);
```

It gives entity framework error 1025. The problem is when I Select the related property Resources, it requires a Func and not Expression. But since entity framework 6 it is possible to use AsQueryable in ICollection properties of an entity, and it works fine if you are using inside a filter method like Where, Any, etc.. But not when you are "Selecting".

I hope this is only a bug so I'm posting as an issue.

**Comment by rowanmiller** [comment]
This doesn't seem like a very common scenario, so it's probably not critical for initial release of EF7. Assigning to @maumar to see if there are any quick wins we can do to unblock this. If it's more involved then we will deal with it post initial release.

**Comment by smitpatel** [comment]
Is this `Include` valid scenario which we would want to fix?

**Comment by ajcvickers** [comment]
@fabiohvp We're not 100% what the intention was when this issue was filed, but in EF Core things have progressed a lot in the last two years. The Select expression inside Include is not supported in EF Core--instead use Include/ThenInclude syntax, as described in the docs. So closing this for now. If you believe there is still an issue with EF Core, then please re-open and provide more details. If this was supposed to be an issue on EF6, then please file a new issue on https://github.com/aspnet/EntityFramework6
